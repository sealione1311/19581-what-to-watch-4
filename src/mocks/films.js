export default [
  {
    id: 1,
    backgroundImage: `https://placeimg.com/1000/550/arch`,
    poster: `https://placeimg.com/270/410/nature`,
    name: `Bohemian Rhapsody`,
    smallImage: `img/bohemian-rhapsody.jpg`,
    genre: `Documentary`,
    released: `2018`,
    description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury.`,
    rating: `8.0`,
    ratingCount: `799`,
    director: `Bryan Singer`,
    starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
    runTime: `1h 39m`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    id: 2,
    backgroundImage: `https://placeimg.com/1000/550/arch`,
    poster: `https://placeimg.com/270/410/nature`,
    name: `Johnny-english`,
    smallImage: `img/johnny-english.jpg`,
    genre: `Comedy`,
    released: `2003`,
    description: `After a sudden attack on MI5, Johnny English, Britain's most confident, yet unintelligent spy, becomes Britain's only spy.`,
    rating: `6.2`,
    ratingCount: `489`,
    director: `Peter Howitt`,
    starring: [`Rowan Atkinson`, `John Malkovich`, `Natalie Imbruglia`],
    runTime: `1h 39m`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 3,
    backgroundImage: `https://placeimg.com/1000/550/arch`,
    poster: `https://placeimg.com/270/410/nature`,
    name: `Macbeth`,
    smallImage: `img/macbeth.jpg`,
    genre: `Drame`,
    released: `2015`,
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
    rating: `7.1`,
    ratingCount: `772`,
    director: `Justin Kurzel`,
    starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
    runTime: `1h 39m`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 4,
    backgroundImage: `https://placeimg.com/1000/550/arch`,
    poster: `https://placeimg.com/270/410/nature`,
    name: `No Country for Old Men`,
    smallImage: `img/no-country-for-old-men.jpg`,
    genre: `Thrillers`,
    released: `2007`,
    description: `Violence and mayhem ensue after a hunter stumbles upon a drug deal gone wrong and more than two million dollars in cash near the Rio Grande.`,
    rating: `8.1`,
    ratingCount: `708`,
    director: `Joel Coen`,
    starring: [`Tommy Lee Jones`, `Javier Bardem`, `Josh Brolin`],
    runTime: `1h 39m`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }
];
