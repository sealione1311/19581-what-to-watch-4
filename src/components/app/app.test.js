import React from "react";
import renderer from "react-test-renderer";
import App from "./app.jsx";
import films from "../../mocks/films.js";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NameSpace from "../../redux/name-space.js";

const mockFilm = {
  id: 1,
  isFavorite: false,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  smallImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  name: `The Grand Budapest Hotel`,
  genre: `Drame`,
  released: 2014,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).
  Zero, a junior lobby boy, becomes Gustave's friend and protege.Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there.
  When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  rating: 8.9,
  ratingCount: 740,
  director: `Wes Anderson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  runTime: 130,
  preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
};

const mockStore = configureStore([]);

describe(`Render App`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      movie: mockFilm,
      films,
      isLoading: false,
      isLoadError: false,
    },
    [NameSpace.STATE]: {
      currentGenre: `All genres`,
      currentMovie: mockFilm,
      displayedFilmsCount: 8,
    },
    [NameSpace.USER]: {
      authorizationStatus: `AUTH`,
      isErrorAuth: false,
      userInfo: {
        id: 1,
        email: `me@gmail.com`,
        name: `12345`,
        avatarUrl: `https://4.react.pages.academy/wtw/me.jpg`,
      }
    },
  });
  it(`Render App`, () => {
    const tree = renderer
    .create(<Provider store={store}>
      <App
        movie={mockFilm}
        films={films}

        login={() => {}}
      /></Provider>, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
