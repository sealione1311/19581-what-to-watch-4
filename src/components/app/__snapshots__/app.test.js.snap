// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Render App 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drame
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Documentary
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drame
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Thrillers
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Sci-fi
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Crime
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          id={1}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="video-player"
              height="175"
              muted="muted"
              poster="img/bohemian-rhapsody.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id={2}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="video-player"
              height="175"
              muted="muted"
              poster="img/johnny-english.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Johnny-english
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id={3}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="video-player"
              height="175"
              muted="muted"
              poster="img/macbeth.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id={4}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="video-player"
              height="175"
              muted="muted"
              poster="img/no-country-for-old-men.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              No Country for Old Men
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id={5}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="video-player"
              height="175"
              muted="muted"
              poster="img/seven-years-in-tibet.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Seven Years in Tibet
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id={6}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="video-player"
              height="175"
              muted="muted"
              poster="img/war-of-the-worlds.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              War of the Worlds
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id={7}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="video-player"
              height="175"
              muted="muted"
              poster="img/orlando.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Orlando
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          id={8}
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              className="video-player"
              height="175"
              muted="muted"
              poster="img/snatch.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Snatch
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
